src/metabase/driver.clj

```diff
@@ -292,11 +292,37 @@
   [date-format-str]
   (.withOffsetParsed ^DateTimeFormatter (tformat/formatter date-format-str)))
 
+(defn- exception-as-return-value
+  "Invoke `f`, return an exception rather than throwing it"
+  [f]
+  (try
+    (f)
+    (catch Exception e
+      e)))
+
+(defn- first-successful-parse
+  "Attempt to parse `time-str` with each of `date-formatters`, returning the first successful parse. If there are no
+  successful parses return the most recent Exception returned."
+  [date-formatters time-str]
+  (loop [[df & more-df] date-formatters]
+    (let [result (exception-as-return-value #(tformat/parse df time-str))]
+      (cond
+        (and (instance? Exception result)
+             (seq more-df))
+        (recur more-df)
+
+        (and (instance? Exception result)
+             (empty? more-df))
+        (throw result)
```

Comment:
so this actually throws the Exception rather than returning it? Docstring suggests otherwise :wink: